src/parser/ast/ProgramNode.h

```diff
@@ -54,7 +54,11 @@ class ProgramNode : public StatementNode {
 
 #ifdef ESCARGOT_DEBUGGER
         if (context->m_breakpointContext->m_breakpointLocations.size() == 0) {
-            insertBreakpoint(context);
+            if (codeBlock->m_isEvalMode) {
+                context->insertBreakpointAt(1, this);
```

Comment:
It's just a question. Why is this part added especially for eval code?

Reply:
Due to pending breakpoints you are able to set a breakpoint into a function created in eval like this:

```js

-> var a = 5;
   eval ("function f(){}");
```
debugger:
```
break f
pending? yes
```

Also whenever the eval code is processed no exact line info exist, also there breakpoint emitting does not happens since there is only a single function, but each parsed script needs at least 1 breakpoint. That's why the manually inserted breakpoint is needed.